### DISCUSSION TITLE
Kwikset HC620 Lock Constant Zwave Chatter

### DISCUSSION THREAD

**Discussion by jdiegmueller** [zwave-js/zwave-js#5566] [General] [answered]
Two times the last few weeks, I've caught my Kwikset HC620 locks misbehaving.  Specifically, one of them will chatter back and forth with the controller over and over and over:

2023-03-13T06:00:00.617Z SERIAL « 0x011a00a800010e119f031d008c2386857a92f97e8db6d6a44900b1a0          (28 bytes)
2023-03-13T06:00:00.620Z SERIAL » [ACK]                                                                   (0x06)
2023-03-13T06:00:00.621Z DRIVER « [Node 014] [REQ] [BridgeApplicationCommand]
                                  │ RSSI: -79 dBm
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 29
                                    └─[SupervisionCCReport]
                                        session id:          6
                                        more updates follow: true
                                        status:              Working
                                        duration:            1s

Once the behavior starts, the above repeats over and over with the sequence number incrementing and then rolling over.  I can still do things on the Zwave mesh including lock/unlock commands, ping, etc.  But the only way I have found to stop this behavior is to pull the battery pack and pop it back in.

I have three of these locks in total and have caught this behavior twice.  Each time it happened to a different lock.

Unfortuntely in both cases, my zwavejs logs rolled over and so I don't know what the "start" of the behavior looks like.  But once it is in this state, it appears to never stop.

I'm attaching today's zwavejs log.  At midnight an automated NVM backup is performed, then it soft resets and immediately picks right back up with the repeating sequence from above.  It looks like I pulled + replaced the battery pack around 2023-03-13T15:41:10.026Z which stopped the constant chatter.

I'm going to keep a closer eye on this and try to catch the "start" of the behavior this week, but figured I'd provide what I know in case anyone has any thoughts or insight in to this behavior.  

[zwavejs_current-ZW1_GONE_CRAZY2.zip](https://github.com/zwave-js/node-zwave-js/files/10959292/zwavejs_current-ZW1_GONE_CRAZY2.zip)

**Comment by AlCalzone** [comment]
The driver should keep logs for 7 days unless you reconfigured that. Maybe the start is in one of the older logfiles?

In any case: what the f...

**Reply by jdiegmueller** [reply]
Unfortunately I didn't notice this one until it was too late, but I did increase log storage to a 90 day circular buffer and restarted so I'll be able to catch the next one.  Will upload here once I catch it!

**Comment by dev77-07** [comment]
Didn't notice this before I opened my issue but I'm experiencing the same thing with my HC620 lock. Started noticing the issue a couple months ago. https://github.com/zwave-js/node-zwave-js/issues/5591 I have my log attached to this issue if that helps at all.

**Reply by jdiegmueller** [reply]
Oddly, this hasn't happened to me again since the original 2 instances.  There must a rare'ish corner case that triggers the behavior.

**Reply by jdiegmueller** [reply]
Happened last night, looks like it was when I left home ~17:35.  Hopefully this is useful.
[zwavejs_2023-04-07.log.gz](https://github.com/zwave-js/node-zwave-js/files/11183491/zwavejs_2023-04-07.log.gz)

**Reply by AlCalzone** [reply]
I have a suspicion:
```
2023-04-07T21:35:43.514Z DRIVER » [Node 028] [REQ] [SendDataBridge]
                                  │ source node id:   1
                                  │ transmit options: 0x25
                                  │ callback id:      123
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 167
                                    └─[SupervisionCCGet]
                                      │ session id:      49
                                      │ request updates: true
                                      └─[DoorLockCCOperationSet]
                                          target mode: Secured
...
2023-04-07T21:35:45.486Z DRIVER « [REQ] [SendDataBridge]
                                    callback id:            123
                                    transmit status:        OK, took 1910 ms
                                    repeater node IDs:      10
                                    routing attempts:       1
                                    protocol & route speed: Z-Wave, 40 kbit/s
                                    ACK RSSI:               -78 dBm
                                    ACK RSSI on repeaters:  -90 dBm
                                    ACK channel no.:        1
                                    TX channel no.:         1
                                    route failed here:      10 -> 28

2023-04-07T21:35:45.554Z DRIVER « [Node 028] [REQ] [BridgeApplicationCommand]
                                  │ RSSI: -77 dBm
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 235
                                    └─[SupervisionCCReport]
                                        session id:          49
                                        more updates follow: true
                                        status:              Working
                                        duration:            2s

2023-04-07T21:35:47.525Z DRIVER « [Node 028] [REQ] [BridgeApplicationCommand]
                                  │ RSSI: -77 dBm
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 236
                                    └─[SupervisionCCReport]
                                        session id:          49
                                        more updates follow: true
                                        status:              Working
                                        duration:            1s

2023-04-07T21:35:48.541Z DRIVER « [Node 028] [REQ] [BridgeApplicationCommand]
                                  │ RSSI: -77 dBm
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 237
                                    └─[SupervisionCCReport]
                                        session id:          49
                                        more updates follow: true
                                        status:              Working
                                        duration:            1s
2023-04-07T21:35:49.553Z SERIAL « 0x011a00a800011c119f03ee00608c44c1297b79ab6daa52dd6d00b24d          (28 bytes)
2023-04-07T21:35:49.559Z SERIAL » [ACK]                                                                   (0x06)
2023-04-07T21:35:49.561Z DRIVER « [Node 028] [REQ] [BridgeApplicationCommand]
                                  │ RSSI: -78 dBm
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 238
                                    └─[SupervisionCCReport]
                                        session id:          49
                                        more updates follow: true
                                        status:              Working
                                        duration:            1s

...
2023-04-07T21:35:51.341Z DRIVER « [Node 028] [REQ] [BridgeApplicationCommand]
                                  │ RSSI: -78 dBm
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 239
                                    └─[SupervisionCCReport]
                                        session id:          49
                                        more updates follow: true
                                        status:              Working
                                        duration:            1s

2023-04-07T21:35:51.601Z DRIVER « [Node 028] [REQ] [BridgeApplicationCommand]
                                  │ RSSI: -79 dBm
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 240
                                    └─[SupervisionCCReport]
                                        session id:          49
                                        more updates follow: true
                                        status:              Working
                                        duration:            1s

2023-04-07T21:35:52.621Z DRIVER « [Node 028] [REQ] [BridgeApplicationCommand]
                                  │ RSSI: -78 dBm
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 241
                                    └─[SupervisionCCReport]
                                        session id:          49
                                        more updates follow: true
                                        status:              Working
                                        duration:            1s
```

The lock first reports that it will take 2s to lock. Then that 1s is left, and then it never stops reporting that.
Almost sounds like they have a loop that will send the working state when `duration > 0`, but they never set duration to 0 when locked.

I'm curious, did anyone ever report this to Kwikset?

**Comment by dev77-07** [comment]
I had opened a ticket with Kwikset but they said issues are usually due to communication issues between the lock and zwave hub or an issue with the hub not the lock. They also said the locks come with the latest firmware and there are no updates. I could update my ticket with your suspicion and see what they say.

**Reply by AlCalzone** [reply]
That would be good. I've updated the log snippet with a few more of the messages. If you tell them, make sure to direct their attention at the ever-increasing sequence number (which means a completely new message is being generated by the device), and the constant session id (meaning it's an update for the same original request).

**Reply by AlCalzone** [reply]
Oh another thing - their Supervision implementation seems to be a bit wonky. Here's another attempt at unlocking the door:
```
# Unlock
2023-04-07T19:17:39.729Z DRIVER » [Node 028] [REQ] [SendDataBridge]
                                  │ source node id:   1
                                  │ transmit options: 0x25
                                  │ callback id:      120
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 166
                                    └─[SupervisionCCGet]
                                      │ session id:      46
                                      │ request updates: true
                                      └─[DoorLockCCOperationSet]
                                          target mode: Unsecured

...

# Door is unlocked, no more updates follow
2023-04-07T19:17:41.749Z DRIVER « [Node 028] [REQ] [BridgeApplicationCommand]
                                  │ RSSI: -78 dBm
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 233
                                    └─[SupervisionCCReport]
                                        session id:          46
                                        more updates follow: false
                                        status:              Success
                                        duration:            0s

...

# Nevermind, here's another update. Expect more updates...
2023-04-07T19:17:41.791Z DRIVER « [Node 028] [REQ] [BridgeApplicationCommand]
                                  │ RSSI: -78 dBm
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 234
                                    └─[SupervisionCCReport]
                                        session id:          46
                                        more updates follow: true
                                        status:              Success
                                        duration:            0s

(no updates follow)
```

**Comment by dev77-07** [comment]
Kwikset is replacing my lock under warranty. I will have to compare firmware versions to see if the new one has newer firmware and see if the issue occurs again. They are also passing my comments to the engineering department, I linked them to this discussion. So hopefully they will see this and look into what appears to be a bug in their firmware.

**Reply by jdiegmueller** [reply]
Thanks for the legwork on the manufacturer side of things.  

I have three of these locks, and all of them have 7.20.0.  Let us (me?) know if you end up with something different.

**Comment by dev77-07** [comment]
Got the new lock today, has the same version as my other one.
![image](https://user-images.githubusercontent.com/58564160/232929345-36d45d30-d16a-49f3-bb54-66fcb984482d.png)

I'll install the lock and see if the same issue happens.

**Reply by jdiegmueller** [reply]
Did Kwikset express any further interest in fixing this in their firmware?  This triggered again for me on April 21 and I didn't catch it and fix it until this morning.  Attaching the zwavejs log in case Kwikset is interested.  This is on a small, separate Zwave network I maintain specifically for my three HC620 locks.

[zwavejs_kwikset-hc620-chatty_20210426A.zip](https://github.com/zwave-js/node-zwave-js/files/11331684/zwavejs_kwikset-hc620-chatty_20210426A.zip)

**Reply by dev77-07** [reply]
They did not. They said they'd pass it on to their engineers and let me know if they had any questions or needed more info and I have not received anything else.

**Comment by yura505** [comment]
@jdiegmueller @yeahme49 Have you resolved the issue? I have the same issue, no idea why. What kind of USB stick do you have? I have HUSBZB-1...

**Reply by dev77-07** [reply]
Nope, still having the issue. I have a ZST10-700 stick.

**Reply by jdiegmueller** [reply]
Same, still happens every couple of days.

For the Zwave network with the 3 locks, I'm using Silicon Labs UZB-7.  Running 7.18.4 currently.

**Comment by yura505** [comment]
@jdiegmueller @yeahme49 When have you started seeing the issue? Recently, or immediately after purchased the lock?

I also opened ticket to Kwikset, but instead of looking to the problem, they recommended b”me to factory reset the lock or replace it, so very disappointed.

**Reply by jdiegmueller** [reply]
I have had the three locks since June 2022, but I didn't notice the misbehavior until maybe 4-6 months ago.  I've only caught it maybe 5 times total.  It's not always the same lock.

**Reply by dev77-07** [reply]
I don't recall when I started having issues but it wasn't immediately after purchase.

I have 2 automations that involve my lock. One unlocks the door when certain people arrive home and another locks the door if it's left unlocked and certain people leave. I don't know for sure but I think my issues started after I created the locking automation. I've disabled the locking automation for the past month or so and haven't had the issue happen. It's hard to tell for sure since it's such a sporadic issue but I'm starting to think the bug is only related to sending lock commands. I never did get anything back from Kwikset's engineering group to see what they had to say. They just sent me the replacement lock and that was the end of things.

**Reply by YuriVizio** [reply]
That sad to hear. I excluded the lock from Z-Wave, because it drains battery and makes all the network unsable. I wanted Z-Wave lock, but using it as regular instead. And yes, no any help from kwikset.

**Reply by jdiegmueller** [reply]
@YuriVizio What firmware are you running on your controller?

**Reply by yura505** [reply]
HUZBZB-1
FW: v4.32
SDK: v6.51.6

**Comment by AlCalzone** [comment, marked as answer]
I had really hoped for them to make some effort at fixing this. As it doesn't seem that way, I'll be disabling Supervision for the HC620 in the next release. This **should** prevent the device from sending these reports, at the downside that we lose some certainty that commands were received and executed.

After updating, the lock needs to be re-interviewed so Z-Wave JS stops sending supervised commands.

Also, I'm putting Kwikset on my "naughty" list:
https://zwave-js.github.io/node-zwave-js/#/getting-started/device-review?id=the-list

**Reply by jdiegmueller** [reply]
@AlCalzone It's probably moot at this point, but: Is there any possibility the controller firmware could have (had) anything to do with this?  

The last time this happened in my environment was April 26.  I can't quite pin down the exact date I did it, but I think I might have updated the controller that has the 3 HC620 locks to 7.18.3 around that date.

**Reply by AlCalzone** [reply]
Maybe this is caused by a missing acknowledgement for the supervision report, causing the lock to re-transmit (which would be okay) and never stop (which is clearly a bug). However you'd need Zniffer traces of the moment this happens to verify.

Not sending an ACK **could** be influenced by the controller firmware, but AFAIK those issues have already been fixed in 7.17.2. It could also just be RF range/interference related.

**Reply by nsheldon** [reply]
Thanks for all the work ion this issue everyone.  I have one of these lock and have not yet seen this issue (I also hardly ever send lock commands via Home Assistant though).

I need to be sure of my lock status at all times. Is there any way to manually re-enable supervision of my HC620 after applying this update? Or perhaps a workaround that could keep my lock status up to date (maybe via automations or something)?

**Reply by AlCalzone** [reply]
@nsheldon Most likely HA is using the `setValue` API which automatically verifies the change after a short delay if Supervision is **not** used. This is the case for Lock commands and several others. You can verify by looking at driver logs on loglevel "debug": https://www.home-assistant.io/integrations/zwave_js#how-to-access-the-z-wave-logs

If this includes a command with DoorLockCCGet after you control the lock, you don't need to do anything.
